plugins/inputs/mavlink/mavlink_utils.go

```diff
@@ -0,0 +1,136 @@
+package mavlink
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/chrisdalke/gomavlib/v3"
+)
+
+// Convert a string from CamelCase to snake_case
+func ConvertToSnakeCase(input string) string {
+	re := regexp.MustCompile(`([a-z0-9])([A-Z])`)
+	snake := re.ReplaceAllString(input, `${1}_${2}`)
+	snake = strings.ToLower(snake)
+	return snake
+}
+
+// Check if a string is in a slice
+func Contains(slice []string, str string) bool {
+	for _, item := range slice {
+		if item == str {
+			return true
+		}
+	}
+	return false
+}
+
+// Convert a Mavlink event into a struct containing Metric data.
+func MavlinkEventFrameToMetric(frm *gomavlib.EventFrame) metricFrameData {
+	out := metricFrameData{}
+	out.tags = make(map[string]string)
+	out.fields = make(map[string]any)
+	out.tags["sys_id"] = strconv.FormatUint(uint64(frm.SystemID()), 10)
+
+	m := frm.Message()
+	t := reflect.TypeOf(m)
+	v := reflect.ValueOf(m)
+	if t.Kind() == reflect.Ptr {
+		t = t.Elem()
+		v = v.Elem()
+	}
+
+	for i := 0; i < t.NumField(); i++ {
+		field := t.Field(i)
+		value := v.Field(i)
+		out.fields[ConvertToSnakeCase(field.Name)] = value.Interface()
```

Comment:
I don't think it is needed to do this conversion, Telegraf is perfectly fine with CamelCase field and metric names.

Reply:
I'd like to keep this around, there's a lot of different conventions in mavlink message naming and not much consistency. The Mavlink docs use capitalized snake case like `MESSAGE_GLOBAL_POSITION_INT`, gomavlib converts that to `MessageGlobalPositionInt`, and some other client libraries use normal snake case like `message_global_position_int`.  

My argument for using lowercase snake case is that, with a lack of one definite standard, I suspect by convention most people will want the outputs to be snake case in their database (Influx, Postgres, etc).